### DISCUSSION TITLE
monai_label WIN 10 installation steps ---> help

### DISCUSSION THREAD

**Discussion by MrMarkusJ** [Project-MONAI/MONAILabel#826] [Q&A] [answered]
Hi,

I am trying to install monai_label on a WIN10 system; 

The installation steps are given already:  https://docs.monai.io/projects/label/en/latest/installation.html#windows

I would like to share my initial experience/problems so that other people can better deal with the installation issues.

- Python: https://www.python.org/downloads/windows/
   python-3.9.11-amd64

- Pytorch
pip install torch==1.9.0+cu111 torchvision==0.10.0+cu111 torchaudio===0.9.0 -f https://download.pytorch.org whl/torch_stable.html

- CUDA
   using NVIDIA Quadro M4000
   Driver Version 516.25
   python -c "import torch; print(torch.cuda.is_available())"
   TRUE

- Install From PyPI
   And from this point on the installation guide was not that clear to me; additional software was needed to be installed
     1. Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools"
         https://visualstudio.microsoft.com/visual-cpp-build-tools/
         MSVC v141 – VS 2017 C++ x64/x86 Buildtools

     2. Windows SDK
         https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/

         (probably these two packages are redundant?, or there might be better choices?)

     3. and an issue regarding the X64 & X86 was encountered
         there was the need to adapt the path:
         https://stackoverflow.com/questions/35215971/lnk1158-cannot-run-rc-exe-x64-visual-studio/

     4. "pip install monailabel"   installation WORKED!
     
- MONAI Label CLI
         command "monailabel --help"  WORKED!

     Downloading Sample Apps or Datasets
     **- unfortunately this was not possible due to the following error:**

      FileNotFoundError: [WinError 3] The system cannot find the specified path: 'C:\\Program Files\\Python39\\monailabel\\sample-apps'

1. Any ideas how to solve this error "FileNotFoundError" and how to proceed with the download of the sample data?
2. Any comments on the undertaken installation steps? 

Thanks!

**Comment by SachidanandAlle** [comment]
> Hi,
> 
> I am trying to install monai_label on a WIN10 system;
> 
> The installation steps are given already: https://docs.monai.io/projects/label/en/latest/installation.html#windows
> 
> I would like to share my initial experience/problems so that other people can better deal with the installation issues.
> 
> * Python: https://www.python.org/downloads/windows/
>   python-3.9.11-amd64
> * Pytorch
>   pip install torch==1.9.0+cu111 torchvision==0.10.0+cu111 torchaudio===0.9.0 -f https://download.pytorch.org whl/torch_stable.html
> * CUDA
>   using NVIDIA Quadro M4000
>   Driver Version 516.25
>   python -c "import torch; print(torch.cuda.is_available())"
>   TRUE
> * Install From PyPI
>   And from this point on the installation guide was not that clear to me; additional software was needed to be installed
>   
>   1. Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools"
>      https://visualstudio.microsoft.com/visual-cpp-build-tools/
>      MSVC v141 – VS 2017 C++ x64/x86 Buildtools
>   2. Windows SDK
>      https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/
>      (probably these two packages are redundant?, or there might be better choices?)
>   3. and an issue regarding the X64 & X86 was encountered
>      there was the need to adapt the path:
>      https://stackoverflow.com/questions/35215971/lnk1158-cannot-run-rc-exe-x64-visual-studio/
>   4. "pip install monailabel"   installation WORKED!
> * MONAI Label CLI
>   command "monailabel --help"  WORKED!
>   Downloading Sample Apps or Datasets
>   **- unfortunately this was not possible due to the following error:**
>   ```
>   FileNotFoundError: [WinError 3] The system cannot find the specified path: 'C:\\Program Files\\Python39\\monailabel\\sample-apps'
>   ```
> 
> 1. Any ideas how to solve this error "FileNotFoundError" and how to proceed with the download of the sample data?
> 2. Any comments on the undertaken installation steps?
> 
> Thanks!

can you install latest release candidate.. instead of stable release..
https://github.com/Project-MONAI/MONAILabel#release-candidate

`pip install monailabel>=0.4*`

**Reply by SachidanandAlle** [reply]
In case of latest version (0.4x onwards) you don't need MSVC tools...
For File error, if you still facing the same issue..  then you can quickly where the python is installed.. normally it tries to find the sample apps folder which are bundled as part of setup package.

> If monailabel install path is not automatically determined, then you can provide explicit install path as: `monailabel apps --prefix ~/.local`

**Reply by MrMarkusJ** [reply]
Hi, 

the "--prefix" option,
e.g. **monailabel apps --prefix C:\Users\.......\AppData\Roaming\Python** 
solved the 

**_FileNotFoundError: [WinError 3]_**

error. 

- next step the Task09_spleen data set was downloaded
- the monai_label server was started:
      **monailabel start_server --app apps/deepedit --studies datasets/Task09_Spleen/imagesTr --conf models deepedit**
- server is active on: http://127.0.0.1:8000
- monai_label plugin was succesfully installed in 3D slicer; connection with the monailable_server was established
- manual labeling using the "scribble" worked; 

- **BUT: the training FAILED:**
  
monai.utils.module.OptionalImportError: from torch.utils.tensorboard import SummaryWriter (Descriptors cannot not be created directly.

If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
If you cannot immediately regenerate your protos, some other possible workarounds are:
 1. Downgrade the protobuf package to 3.20.x or lower.
 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates).
For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies
[2022-06-08 16:07:02,643] [21860] [ThreadPoolExecutor-1_0] [INFO] (monailabel.utils.async_tasks.utils:76) - Return code: 1

How to proceed?

Thanks!

**Comment by SachidanandAlle** [comment]
https://github.com/Project-MONAI/MONAILabel/blob/main/requirements.txt#L31

These problem is solved by downgrade..

**Comment by SachidanandAlle** [comment, marked as answer]
R u not using 0.4 versions?

pip install monailabel>=0.4*

**Reply by MrMarkusJ** [reply]
Hi,

the command 

- pip install monailabel>=0.4*

resulted in a Syntax Error (?no idea why; on WIN10).

What worked was to install the weekly release:

- pip install monailabel-weekly

now installed:

- Shapely-1.8.1.post1 

- diskcache-5.4.0 

- girder-client-3.1.8 

- monai-0.9.0rc3 

- monailabel-weekly-0.4.dev2223 

- numpymaxflow-0.0.2 

- opencv-python-headless-4.5.5.64 

- protobuf-3.20.1 

- pynrrd-0.4.2 

- uvicorn-0.17.5

Result:
1. The monailabel_SERVER started: monailabel start_server --app apps/deepedit --studies datasets/Task09_Spleen/imagesTr --conf models deepedit

2. Training started

3. ERROR: CUDA out of memory ( 8 GiB total capacity available, but not enough; using NVIDIA M4000 8GB).

Will search on the MONAI-forum for an answer.

**Reply by SachidanandAlle** [reply]
I just tried on my windows.. I used powershell
I think you used basic command prompt.. in that case you can use direct latest version:
https://pypi.org/project/monailabel/0.4.0rc6/

Before that you should uninstall monailabel-weekly (they conflict each other)

Regarding training, you need 16GB GPU for deepedit.. you can try segmentation_spleen model.  it might work on smaller GPU

**Reply by MrMarkusJ** [reply]
Hi,
still working with the weekly release version.

Trying now the "radiology-app"

- monailabel apps --download --name radiology --output apps 

And starting the monailabel_server using "segmentation":

- monailabel start_server --app apps/radiology --studies datasets/Task09_Spleen/imagesTr --conf models segmentation

**RESULT:**

- GPU memory of 8 GB is sufficient to train on the 512x512x96 data sets "Task09_Spleen"
